lib/slack/rtm.ex

```diff
@@ -10,15 +10,26 @@ defmodule Slack.Rtm do
   @moduledoc false
 
   def start(token) do
-    url = Application.get_env(:slack, :url, "https://slack.com") <> "/api/rtm.start?token="
+    slack_url(token)
+    |> HTTPoison.get()
+    |> handle_response()
+  end
 
-    case HTTPoison.get(url <> token) do
-      {:ok, %HTTPoison.Response{body: body}} ->
-        case JSX.decode(body, [{:labels, :atom}]) do
-          {:ok, json}       -> {:ok, json}
-          {:error, reason}  -> {:error, %JSX.DecodeError{reason: reason, string: body}}
-        end
-      {:error, reason} -> {:error, reason}
+  defp handle_response({:ok, %HTTPoison.Response{body: body}}) do
+    with {:ok, json} <- JSX.decode(body, [{:labels, :atom}]),
```

Comment:
I'm not sure `with` makes the code easier to read here. Thoughts on extracting a function to decode the JSON and make sure that Slack returned `ok: true`?

I think this function could then look like

```
defp handle_response({:ok, %HTTPoison.Response{body: body}}) do
  decode_and_handle_slack_response(body)
end
```

Which could return `{:error, reason}` if the parsing failed, or if `ok: false`.  (and `{:ok, json}` if it was successful)

Another thought is that we could *maybe* do `JSX.decode!`, because Slack sending us down invalid JSON is exceptional, and could be a good reason to 'let it crash".